seeds/index.js

```diff
@@ -9,41 +9,23 @@ const rops = require('./tables/rops');
 const dataExports = require('./tables/exports');
 const enforcementCases = require('./tables/enforcement-cases');
 
-exports.seed = knex => {
-  return Promise.resolve()
-    .then(() => dataExports.delete(knex))
-    .then(() => enforcementCases.delete(knex))
-    .then(() => knex('establishment_merge_log').del())
-    .then(() => knex('document_cache').del())
-    .then(() => knex('changelog').del())
-    .then(() => knex('notifications').del())
-    .then(() => knex('exports').del())
-    .then(() => knex('training_pils').del())
-    .then(() => knex('training_courses').del())
-    .then(() => knex('project_profiles').del())
-    .then(() => knex('retrospective_assessments').del())
-    .then(() => knex('procedures').del())
-    .then(() => rops.delete(knex))
-    .then(() => additionalAvailability.delete(knex))
-    .then(() => projectVersions.delete(knex))
-    .then(() => projects.delete(knex))
-    .then(() => knex('place_roles').del())
-    .then(() => places.delete(knex))
-    .then(() => knex('certificates').del())
-    .then(() => knex('exemptions').del())
-    .then(() => knex('email_preferences').del())
-    .then(() => profiles.delete(knex))
-    .then(() => knex('authorisations').del())
-    .then(() => establishments.delete(knex))
+exports.seed = async knex => {
+  const tables = await knex.select('table_name')
+    .from('information_schema.tables')
+    .whereRaw('table_schema = current_schema()')
+    .where('table_catalog', knex.client.database())
+    .then(results => results.map(r => r.tableName).filter(tableName => !tableName.includes('knex_')));
 
-    .then(() => establishments.populate(knex))
-    .then(() => profiles.populate(knex))
-    .then(() => places.populate(knex))
-    .then(() => projects.populate(knex))
-    .then(() => projectVersions.populate(knex))
-    .then(() => additionalAvailability.populate(knex))
-    .then(() => trainingCourses.populate(knex))
-    .then(() => rops.populate(knex))
-    .then(() => enforcementCases.populate(knex))
-    .then(() => dataExports.populate(knex));
+  await Promise.all(tables.map(table => knex.raw(`TRUNCATE TABLE ${table} CASCADE`)));
```

Comment:
This seems to work fine locally, but it will execute them all in parallel... which may or may not be a good idea performance-wise with Postgres, as the CASCADE will truncate all related tables as well, meaning some tables may get truncated several times.

However, as we only run the full seed sparingly, it's probably not worth worrying about compared to the convenience of not having to specify the right order of truncates/deletes.

Reply:
As they are running in parallel, is there any chance of a deadlock where the cascade is trying to truncate tables truncated by another statement or cascade?

Reply:
100% not concerned about performance while it's O(seconds)